### DISCUSSION TITLE
Using with netlify functions

### DISCUSSION THREAD

**Discussion by s-kris** [nextauthjs/next-auth#1477] [Ideas]
I deployed my nextjs + nextauth app in netlify but because of the 10s execution limit, logging via social accounts is not possible. Everytime, 'function execution time exceeded' error is thrown after redirect. 

Is there way to solve this? 

Note:
The nextjs-netlify plugin converts any api route that ends with "-background" to a background function and it will have 15m execution time.

**Comment by StefanSelfTaught** [comment]
Hmm on vercel the execution limit is also 10 seconds for the free plan and it works for me.  Maybe you can try some things:

-  Deploy your functions to a better region
-  As far as I understood you can get a 26 seconds execution limit, you just have to contact the netfliy team
-  Deploy your function directly to AWS Lambda
-  Use background functions

**Reply by s-kris** [reply]
Hi @StefanSelfTaught 

I'm already using background functions for certain api routes.
But for nextjs-plugin, we need to attach "-background" to the route's name as mentioned in OP.  So, is there a way to add this to next-auth's routes?
